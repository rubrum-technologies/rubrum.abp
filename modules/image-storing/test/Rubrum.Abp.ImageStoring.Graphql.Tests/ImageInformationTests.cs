using CookieCrumble;
using HotChocolate.Types.Relay;
using Shouldly;
using Xunit;
using static Rubrum.Abp.ImageStoring.ImageStoringTestConstants;

namespace Rubrum.Abp.ImageStoring;

public sealed class ImageInformationTests : ImageStoringGraphqlTestBase
{
    private readonly IImageInformationRepository _repository;
    private readonly IIdSerializer _idSerializer;

    public ImageInformationTests()
    {
        _repository = GetRequiredService<IImageInformationRepository>();
        _idSerializer = GetRequiredService<IIdSerializer>();
    }

    [Fact]
    public async Task FetchUrlById()
    {
        var image = await _repository.GetAsync(JpegId);
        var id = _idSerializer.Serialize(null, ImageInformationConstants.TypeName, image.Id);

        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  imageUrl(id: "{{id}}")
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchUrlsByTag()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  imageUrlsByTag(tag: "{{ImageTag}}")
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchById()
    {
        var id = _idSerializer.Serialize(null, ImageInformationConstants.TypeName, GifId);

        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  image(id: "{{id}}") {
                    id
                    tag
                    fileName
                    systemFileName
                    isDisposable
                    entityVersion
                    url
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchByTag()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  imagesByTag(tag: "{{ImageTag}}"){
                    id
                    tag
                    fileName
                    systemFileName
                    isDisposable
                    entityVersion
                    url
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }
}
