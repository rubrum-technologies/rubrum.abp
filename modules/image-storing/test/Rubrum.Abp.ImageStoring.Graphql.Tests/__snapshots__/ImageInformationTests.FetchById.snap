{
  "data": {
    "image": {
      "id": "SW1hZ2VJbmZvcm1hdGlvbgpnY2ViNTA3MGJhNWM5NDMxYzhiOGEzYzBlZjc3YWUwMzM=",
      "tag": null,
      "fileName": "6.gif",
      "systemFileName": "ceb5070b-a5c9-431c-8b8a-3c0ef77ae033.webp",
      "isDisposable": false,
      "entityVersion": 0,
      "url": "/api/image-storing/images/download/ceb5070b-a5c9-431c-8b8a-3c0ef77ae033"
    }
  }
}